pkg/kapp/cmd/core/deps_factory.go

```diff
@@ -83,6 +87,29 @@ func (f *DepsFactoryImpl) CoreClient() (kubernetes.Interface, error) {
 	return clientset, nil
 }
 
+func (f *DepsFactoryImpl) RESTMapper() (meta.RESTMapper, error) {
+	config, err := f.configFactory.RESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	disc, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	groupResources, err := restmapper.GetAPIGroupResources(disc)
+	if err != nil {
+		return nil, err
+	}
+
+	mapper := restmapper.NewDiscoveryRESTMapper(groupResources)
```

Comment:
Generally you want to use the deferred mapper so you don't send 300 requests to the server on startup.

Reply:
Deferred mapper used in https://github.com/carvel-dev/kapp/pull/887/commits/feea616b47b08fef28ad4f0fbd565c600bb37575